packages/tools/src/tools/annotation/PlanarFreehandROITool.ts

```diff
@@ -0,0 +1,446 @@
+    let previousPoint = viewport.worldToCanvas(points[0]);
+
+    for (let i = 1; i < points.length; i++) {
+      const p1 = previousPoint;
+      const p2 = viewport.worldToCanvas(points[i]);
+
+      const distance = pointCanProjectOnLine(canvasCoords, p1, p2, proximity);
+
+      if (distance) {
+        return true;
+      }
+
+      previousPoint = p2;
+    }
+
+    if (annotation.data.isOpenContour) {
+      // Contour is open, don't check last point to first point.
+      return false;
+    }
+
+    // check last point to first point
+    const pStart = viewport.worldToCanvas(points[0]);
+    const pEnd = viewport.worldToCanvas(points[points.length - 1]);
+
+    const distance = pointCanProjectOnLine(
+      canvasCoords,
+      pStart,
+      pEnd,
+      proximity
+    );
+
+    if (distance) {
+      return true;
+    }
+
+    return false;
+  };
+
+  cancel = (element: HTMLDivElement): void => {
+    // TODO CANCEL
+  };
+
+  /**
+   * Triggers an annotation modified event.
+   */
+  triggerAnnotationModified = (
+    annotation: PlanarFreehandROIAnnotation,
+    enabledElement: Types.IEnabledElement
+  ): void => {
+    const { viewportId, renderingEngineId } = enabledElement;
+    // Dispatching annotation modified
+    const eventType = Events.ANNOTATION_MODIFIED;
+
+    const eventDetail: AnnotationModifiedEventDetail = {
+      annotation,
+      viewportId,
+      renderingEngineId,
+    };
+    triggerEvent(eventTarget, eventType, eventDetail);
+  };
+
+  /**
+   * Triggers an annotation completed event.
+   */
+  triggerAnnotationCompleted = (
+    annotation: PlanarFreehandROIAnnotation
+  ): void => {
+    const eventType = Events.ANNOTATION_COMPLETED;
+
+    const eventDetail: AnnotationCompletedEventDetail = {
+      annotation,
+    };
+
+    triggerEvent(eventTarget, eventType, eventDetail);
+  };
+
+  /**
+   * Draws the `PlanarFreehandROIAnnotation`s at each request animation frame.
+   *
+   * @param enabledElement - The Cornerstone's enabledElement.
+   * @param svgDrawingHelper - The svgDrawingHelper providing the context for drawing.
+   */
+  renderAnnotation = (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: any
+  ): void => {
+    const { viewport } = enabledElement;
+    const { element } = viewport;
+
+    let annotations = <PlanarFreehandROIAnnotation[]>(
+      getAnnotations(element, this.getToolName())
+    );
+
+    // Todo: We don't need this anymore, filtering happens in triggerAnnotationRender
+    if (!annotations?.length) {
+      return;
+    }
+
+    annotations = this.filterInteractableAnnotationsForElement(
+      element,
+      annotations
+    ) as PlanarFreehandROIAnnotation[];
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    const isDrawing = this.isDrawing;
+    const isEditingOpen = this.isEditingOpen;
+    const isEditingClosed = this.isEditingClosed;
+
+    if (!(isDrawing || isEditingOpen || isEditingClosed)) {
```

Comment:
Which case it is? so it is not drawing, and not editing? does it means it is on hover? can we add that comment?

Reply:
The renderAnnotation callback renders all tools, this just says render them all as tooldata, and there will be no previews, which need different render paths, will add a comment.